nuget/helpers/lib/NuGetUpdater/NuGetUpdater.Core.Test/Update/UpdateWorkerTests.PackagesConfig.cs

```diff
@@ -1663,67 +1666,166 @@ public async Task PackagesConfigUpdateIsNotThwartedBy_VSToolsPath_PropertyBeingS
         [Fact]
         public async Task PackageCanBeUpdatedWhenAnotherInstalledPackageHasBeenDelisted()
         {
-            // updating one package (Newtonsoft.Json) when another installed package (FSharp.Core/5.0.3-beta.21369.4) has been delisted
-            await TestUpdateForProject("Newtonsoft.Json", "7.0.1", "13.0.1",
+            // updating one package (Some.Package) when another installed package (Delisted.Package/5.0.0) has been delisted
+            // this test can't be faked with a local package source and requires an HTTP endpoint; the important part is
+            // the `"listed": false` in the registration index
+            static (int, byte[]) TestHttpHandler(string uriString)
+            {
+                var uri = new Uri(uriString, UriKind.Absolute);
+                var baseUrl = $"{uri.Scheme}://{uri.Host}:{uri.Port}";
+                var path = uri.PathAndQuery;
+                var x = path;
+                return uri.PathAndQuery switch
+                {
+                    "/index.json" => (200, Encoding.UTF8.GetBytes($$"""
+                    {
+                        "version": "3.0.0",
+                        "resources": [
+                            {
+                                "@id": "{{baseUrl}}/download",
+                                "@type": "PackageBaseAddress/3.0.0"
+                            },
+                            {
+                                "@id": "{{baseUrl}}/query",
+                                "@type": "SearchQueryService"
+                            },
+                            {
+                                "@id": "{{baseUrl}}/registrations",
+                                "@type": "RegistrationsBaseUrl"
+                            }
+                        ]
+                    }
+                    """)),
+                    "/registrations/delisted.package/index.json" => (200, Encoding.UTF8.GetBytes($$"""
+                        {
+                            "count": 1,
+                            "items": [
+                                {
+                                    "lower": "5.0.0",
+                                    "upper": "5.0.0",
+                                    "items": [
+                                        {
+                                            "catalogEntry": {
+                                                "id": "Delisted.Package",
+                                                "listed": false,
```

Comment:
This line is what requires this to be a network test; there's no way to fake this with a local packge feed.